strawberryfields/backends/fockbackend/ops.py

```diff
@@ -231,8 +234,14 @@ def apply_gate_BLAS(mat, state, pure, modes, n, trunc):
     transpose_list = [2*i for i in range(size)] + [2*i + 1 for i in range(size)]
     matview = np.transpose(mat, transpose_list).reshape((dim, dim))
 
+    # checks if matview is diagonal and, if so, use faster contractions
+    diag = np.all(matview == np.diag(np.diagonal(matview)))
+
     if pure:
         if n == 1:
+            if diag:
+                mat_diag = mat.diagonal()
+                return np.multiply(mat_diag, state)
```

Comment:
Nice!

Reply:
Does this cause specific speedup in certain use-cases?

Reply:
yup, for the Kerr gate and the rotation gate since they are diagonal.

Reply:
Awesome 🎆